### PR TITLE
Add upstream WPT sync webhook server

### PR DESCRIPTION
**Pull request by jdm**
This is flagrantly based off of the setup for intermittent-tracker. I've added the appropriate secret to servo-master1's pillar, and I've also cloned the web-platform-tests repo to /srv/web-platform-tests.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/733)

### CODE REVIEW

**Comment by jdm** [comment]
Added documentation.

File: .travis/test_pillars/homu.sls [hunk 1]
```diff
@@ -8,3 +8,4 @@ homu:
   'buildbot-http-pass': 'TEST_HOMU_BUILDBOT_HTTP_PASS'
   'travis-ci-token': 'TEST_HOMU_TRAVIS_CI_TOKEN'
   'web-secret': 'TEST_HOMU_WEB_SECRET'
+  'upstream-wpt-sync-token': 'TEST_UPSTREAM_WPT_SYNC_TOKEN'
```
Comment (aneeshusa) [inline .travis/test_pillars/homu.sls]: Add a trailing newline here.

File: upstream-wpt-webhook/map.jinja [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{%
+  set webhook = {
+    'rev': 'a152b5d5bcc107ec8bcc3f326c6b1b62d87a1e04'
+  }
+%}
```
Comment (aneeshusa) [inline upstream-wpt-webhook/map.jinja]: Add a trailing newline here as well.

File: upstream-wpt-webhook/files/wpt-webhook.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+exec /home/servo/upstream-wpt-sync-webhook//_venv/bin/upstream_wpt_webhook
```
Comment (aneeshusa) [inline upstream-wpt-webhook/files/wpt-webhook.conf]: The double slash looks suspicious here.

File: .travis/test_pillars/homu.sls (hunk 1 above)
Comment (aneeshusa) [reply .travis/test_pillars/homu.sls]: Also, make a new file for this instead of putting it into the `homu.sls` file.

File: Vagrantfile [hunk 4]
```diff
@@ -63,6 +64,7 @@ Vagrant.configure(2) do |config|
         # testing out changes locally.
         machine.vm.synced_folder dir, '/tmp/salt-testing-root/saltfs'
         machine.vm.synced_folder test_pillars_path, '/srv/pillar'
+        machine.vm.synced_folder test_wpt_path, '/srv/web-platform-tests'
```
Comment (aneeshusa) [inline Vagrantfile]: Why do this? Seems it would be easier to keep this with the rest of the app's files.

File: upstream-wpt-webhook/init.sls [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - name: /home/servo/upstream-wpt-sync-webhook/_venv
+    - venv_bin: virtualenv-3.5
+    - python: python3
+    - system_site_packages: False
+    - require:
+      - pkg: python3
+      - pip: virtualenv
+  pip.installed:
+    - pkgs:
+      - git+https://github.com/servo-automation/upstream-wpt-sync-webhook@{{ webhook.rev }}
```
Comment (aneeshusa) [inline upstream-wpt-webhook/init.sls]: It would be nice if you publish to PyPI - the Salt state does much better with PyPI packages than git packages.

Comment (aneeshusa) [reply upstream-wpt-webhook/init.sls]: Never mind, no longer needed as #593 is fixed.

**Comment by jdm** [comment]
Review comments addressed.

**Comment by metajack** [comment]
squash the fixups and r=me

**Pull request merged**